tensorboard/main.py

```diff
@@ -279,9 +272,26 @@ def ConstructDebuggerPluginWithGrpcPort(context):
         projector_plugin.ProjectorPlugin,
         text_plugin.TextPlugin,
         profile_plugin.ProfilePlugin,
-        ConstructDebuggerPluginWithGrpcPort,
     ]
 
+    def ConstructDebuggerPluginWithGrpcPort(context):
+      try:
+        # pylint: disable=line-too-long,g-import-not-at-top
+        from tensorboard.plugins.debugger import debugger_plugin as debugger_plugin_lib
+        # pylint: enable=line-too-long,g-import-not-at-top
+      except ImportError as err:
```

Comment:
Reraising like this destroys the traceback. Observe: the program
```python
import threading
import six
import sys


def a():
  return b()
def b():
  return c()
def c():
  raise ValueError('Oh, dear!')

def with_explicit_raise():
  try:
    a()
  except ValueError as e:
    raise ValueError(e.message + ' There was a problem.')


def with_six_reraise():
  try:
    a()
  except ValueError as e:
    (unused_type, unused_value, traceback) = sys.exc_info()
    six.reraise(
        ValueError,
        ValueError(e.message + ' There was a problem.'),
        traceback)


if __name__ == '__main__':
  for fn in (with_explicit_raise, with_six_reraise):
    t = threading.Thread(name=fn.__name__, target=fn)
    t.start()
```
yields
```
Exception in thread with_explicit_raise:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/tmp/z.py", line 17, in with_explicit_raise
    raise ValueError(e.message + ' There was a problem.')
ValueError: Oh, dear! There was a problem.

Exception in thread with_six_reraise:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/tmp/z.py", line 28, in with_six_reraise
    traceback)
  File "/tmp/z.py", line 22, in with_six_reraise
    a()
  File "/tmp/z.py", line 7, in a
    return b()
  File "/tmp/z.py", line 9, in b
    return c()
  File "/tmp/z.py", line 11, in c
    raise ValueError('Oh, dear!')
ValueError: Oh, dear! There was a problem.
```
Try using the latter formulation instead. (But with `ImportError`s instead of `ValueError`s.)

Reply:
Thanks. Done.